### PR TITLE
Fix language-map.json build script

### PR DESCRIPTION
**Pull request by zm-cttae**
- Search for extension manifests in microsoft/vscode instead of microsoft (half the manifests, more languages).
- ~~Add logic to try match contributed filenames/extensions to their icon using the existing icon map.~~
- ~~Abstract the icon name logic in the build script.~~

Test case:
- v1.7.3 will throw an error in `processLanguageContribution` because `lang` isn't reassigned back
- https://github.com/microsoft/vscode/blob/1.74.0/extensions/shellscript/package.json#L28-L69
  Once fixed and run, any matches to `console` will disappear in `language-map.json`

### CODE REVIEW

File: scripts/build-languages.js [hunk 1]
```diff
@@ -226,3 +185,22 @@ function generateLanguageMap() {
   console.log('[7/7] Deleting language contribution cache.');
   rimraf.sync(vsDataPath);
 }
+
+function handleIconRemapping(language, overrideType, contributionType, value) {
```
Comment (meche-gh) [inline scripts/build-languages.js:189]: @Claudiohbsantos

- In PR #10 it was decided to keep this extension lean and rely on the icon pack
- But that PR added a remap/overrides JSON (e.g. to make `*.npmrc` use `npm` icon)
- The overrides mostly don't make sense e.g. stops the settings and properties icons from appearing on generic ignore/setup files
  For example, `test.gitconfig` gets the git icon but doesn't apply any configuration in git

Can I remove those overrides from `language-map.json` and then uncheck `language-map.json`, `remap.json`?

<details>
<summary>This is the diff in `language-map.json`</summary>

```diff
diff --git a/src/language-map.json b/src/language-map.json
index 6cc164e..89ac1cb 100644
--- a/src/language-map.json
+++ b/src/language-map.json
@@ -45,7 +45,7 @@
     "dsql": "database",
     "dtml": "xml",
     "ebuild": "console",
-    "editorconfig": "editorconfig",
+    "editorconfig": "settings",
     "edn": "clojure",
     "ent": "xml",
     "es6": "javascript",
@@ -59,9 +59,11 @@
     "fxml": "xml",
     "gemspec": "ruby",
     "geojson": "json",
-    "gitattributes": "git",
-    "gitconfig": "git",
-    "gitmodules": "git",
+    "gitattributes": "settings",
+    "gitconfig": "settings",
+    "gitignore": "git",
+    "gitignore_global": "git",
+    "gitmodules": "settings",
     "gvy": "groovy",
     "gyp": "python",
     "gypi": "python",
@@ -102,7 +104,8 @@
     "mod": "xml",
     "mts": "typescript",
     "nf": "groovy",
-    "npmrc": "npm",
+    "npmignore": "git",
+    "npmrc": "settings",
     "nqp": "perl",
     "nuspec": "xml",
     "opml": "xml",
@@ -192,10 +195,8 @@
   },
   "fileNames": {
     ".devcontainer.json": "json",
-    ".env": "tune",
     ".envrc": "console",
     ".hushlogin": "console",
-    ".vscodeignore": "vscode",
     "APKBUILD": "console",
     "COMMIT_EDITMSG": "git",
     "Containerfile": "docker",
@@ -206,8 +207,8 @@
     "Makefile": "settings",
     "OCamlMakefile": "settings",
     "PKGBUILD": "console",
-    "SConscript": "python-misc",
-    "SConstruct": "python-misc",
+    "SConscript": "python",
+    "SConstruct": "python",
     "appraisals": "ruby",
     "argv.json": "json",
     "bashrc_Apple_Terminal": "console",
@@ -221,7 +222,7 @@
     "devcontainer.json": "json",
     "extensions.json": "json",
     "git-rebase-todo": "git",
-    "gitconfig": "git",
+    "gitconfig": "settings",
     "guardfile": "ruby",
     "gymfile": "ruby",
     "hobofile": "ruby",
```
</details>

Comment (meche-gh) [reply scripts/build-languages.js:189]: thoughts @shivapoudel? the PR author deleted their account unfortunately

Comment (meche-gh) [reply scripts/build-languages.js]: I went ahead and removed it in [`866254f`](https://github.com/Claudiohbsantos/github-material-icons-extension/pull/49/commits/866254f115bb9745db5592bfc2e490d827ab6619).

**Comment by Claudiohbsantos** [comment]
@zm-cttae I know I'm already a couple of months late but just wanted to bump this PR so you see that I saw it. I will need to dig a little bit into what these changes would mean so it'll take me a little bit to have a proper review but overall I'm not against these changes.

**Comment by zm-cttae** [comment]
Cool. The big change is https://github.com/Claudiohbsantos/github-material-icons-extension/pull/49#discussion_r1045286088 which brings us closer to the upstream extension in behaviour

**Pull request merged**